animations/animation-delay-changed.html is randomly failing

fujii opened the pull request:
#### e244db9b198977e223e12ea4acbb236e55d891e9
<pre>
animations/animation-delay-changed.html is randomly failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=111612">https://bugs.webkit.org/show_bug.cgi?id=111612</a>

Reviewed by Don Olmstead.

In this test case, both box1 and box2 move from 0 to 200, and go back
to 0 when the animations finish. The test case checks if box1 position
is around 200. However, it&apos;s the end of the animation. Thus, the
position randomly results in around 200 or 0.

On the other hand, the test case also checks if box2 position is
around 100. It&apos;s the middle of the animation. Do the same for box1.
Check the position of box1 at the middle of the animation.

* LayoutTests/animations/animation-delay-changed-expected.txt:
* LayoutTests/animations/animation-delay-changed.html:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/wincairo/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/268812@main">https://commits.webkit.org/268812@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0810fc01f112bfac98245d762564499f65561844

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19682 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20104 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20721 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21576 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18404 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23365 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20247 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/19979 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19901 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19908 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17118 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22430 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17088 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17909 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24197 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18155 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18083 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22178 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18685 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/15838 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17834 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/17745 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22188 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2561 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18510 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on LayoutTests/animations/animation-delay-changed.html:
```diff
@@ -47,7 +47,7 @@
       const expectedValues = [
         // [animation-name, time, element-id, property, expected-value, tolerance]
-        [null, 1.0, "box1", "left",  200, 10],
+        [null, 0.5, "box1", "left",  100, 10],
```
I don't understand the change. Why is it less time-sensitive to check the position halfway through the animation instead of at the end? Also, why don't we need to change the similar test for `box2`.

fujii replied:
Both `box1` and `box2` animate from 0 to 200.
`box2` test is already testing at 100. It's the middle of the animation.
`box2` animation starts 0.5s after `box1` starts.

The problem of testing at the end of the animation is the left position goes back to 0 when the animation ends.
So, it randomly ends in 0 or 200 at the end of the animation.

fujii replied:
Revided the commit message.

donny-dont commented on LayoutTests/animations/animation-delay-changed.html:
```diff
@@ -47,7 +47,7 @@
 
       const expectedValues = [
         // [animation-name, time, element-id, property, expected-value, tolerance]
-        [null, 1.0, "box1", "left",  200, 10],
```
Would the value always be `200` if we just added `-webkit-animation-fill-mode: forwards`?

It seems like this test is checking if it ended but the behavior is if the check is done at the end its already snapped back to `0`.

fujii replied:
Yes, adding `-webkit-animation-fill-mode:forwards` also resolves the problem.
But, I think my approach is better than yours.
This test is checking if box1 is exactly 0.5s earlier than box2. With your approach, box1 can be more earlier.

donny-dont reviewed (commented):
Not sure this is the correct change to make

fujii committed:
animations/animation-delay-changed.html is randomly failing
https://bugs.webkit.org/show_bug.cgi?id=111612

Reviewed by Don Olmstead.

In this test case, both box1 and box2 move from 0 to 200, and go back
to 0 when the animations finish. The test case checks if box1 position
is around 200. However, it's the end of the animation. Thus, the
position randomly results in around 200 or 0.

On the other hand, the test case also checks if box2 position is
around 100. It's the middle of the animation. Do the same for box1.
Check the position of box1 at the middle of the animation.

* LayoutTests/animations/animation-delay-changed-expected.txt:
* LayoutTests/animations/animation-delay-changed.html:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/wincairo/TestExpectations:

Canonical link: https://commits.webkit.org/268812@main
